src/v/config/configuration.cc

```diff
@@ -2061,16 +2061,15 @@ configuration::configuration()
       "past the local retention limit, and will be trimmed automatically as "
       "storage reaches the configured target size.",
       {.needs_restart = needs_restart::no, .visibility = visibility::user},
-      false,
-      property<bool>::noop_validator,
-      legacy_default<bool>(true, legacy_version{9}))
+      false)
```

Comment:
>In v23.3 ... regardless of the upgrade history of the cluster

Is this sentence meant to indicate the behavior changed in this commit? If so, could you remove the mention of v23.3? It's a bit confusing without more context, since the dev branch is staging for v24.1